src/utilities/getSmallestNumber/getSmallestNumber.test.js

```diff
@@ -0,0 +1,126 @@
+      });
+    });
+  });
+
+  describe('when provided with invalid inputs', () => {
+    describe('and input is not an array', () => {
+      it('should throw error for null', () => {
+        expect(() => getSmallestNumber(null)).toThrow('Input must be an array');
+      });
+
+      it('should throw error for undefined', () => {
+        expect(() => getSmallestNumber(undefined)).toThrow('Input must be an array');
+      });
+
+      it('should throw error for string', () => {
+        expect(() => getSmallestNumber('123')).toThrow('Input must be an array');
+      });
+
+      it('should throw error for number', () => {
+        expect(() => getSmallestNumber(123)).toThrow('Input must be an array');
+      });
+
+      it('should throw error for object', () => {
+        expect(() => getSmallestNumber({})).toThrow('Input must be an array');
+      });
+
+      it('should throw error for boolean', () => {
+        expect(() => getSmallestNumber(true)).toThrow('Input must be an array');
+      });
+    });
+
+    describe('and array is empty', () => {
+      it('should throw error for empty array', () => {
+        expect(() => getSmallestNumber([])).toThrow('Array cannot be empty');
+      });
+    });
+
+    describe('and array contains non-numeric values', () => {
+      it('should throw error for array with strings', () => {
+        expect(() => getSmallestNumber([1, '2', 3])).toThrow('All array elements must be valid numbers');
+      });
+
+      it('should throw error for array with null', () => {
+        expect(() => getSmallestNumber([1, null, 3])).toThrow('All array elements must be valid numbers');
+      });
+
+      it('should throw error for array with undefined', () => {
+        expect(() => getSmallestNumber([1, undefined, 3])).toThrow('All array elements must be valid numbers');
+      });
+
+      it('should throw error for array with boolean', () => {
+        expect(() => getSmallestNumber([1, true, 3])).toThrow('All array elements must be valid numbers');
+      });
+
+      it('should throw error for array with objects', () => {
+        expect(() => getSmallestNumber([1, {}, 3])).toThrow('All array elements must be valid numbers');
+      });
+
+      it('should throw error for array with arrays', () => {
+        expect(() => getSmallestNumber([1, [2], 3])).toThrow('All array elements must be valid numbers');
+      });
+
+      it('should throw error for array with NaN', () => {
+        expect(() => getSmallestNumber([1, NaN, 3])).toThrow('All array elements must be valid numbers');
+      });
+    });
+  });
+
+  describe('when checking if the original array is not mutated', () => {
+    it('should not mutate the original array', () => {
+      const originalArray = [5, 2, 8, 1, 9];
+      const arrayClone = [...originalArray];
+      
+      getSmallestNumber(originalArray);
+      
+      expect(originalArray).toEqual(arrayClone);
+      expect(originalArray.length).toBe(arrayClone.length);
+    });
+  });
```

Comment:
This `describe` block is not really necessary. I would remove it and leave just the `it`

```
it('should not mutate the original array', () => {
  const originalArray = [5, 2, 8, 1, 9];
  const arrayClone = [...originalArray];

  getSmallestNumber(originalArray);

  expect(originalArray).toEqual(arrayClone);
  expect(originalArray.length).toBe(arrayClone.length);
});
```